sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1556,6 +1556,14 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       df.where($"city".contains(new java.lang.Character('A'))),
       Seq(Row("Amsterdam")))
   }
+
+  test("SPARK-25942: typed aggregation on primitive data") {
```

Comment:
how was this bug introduced?

Reply:
It should not be introduced recently. I think we encode primitive data as a `value` field at beginning. 

Because `AppendColumns` takes serializers' named expressions as `newColumns`. For primitive data, it conflicts with original data attribute `value`.